diff --git a/server/TetherController.cpp b/server/TetherController.cpp
index 0c5b6bfe..d9439be6 100644
--- a/server/TetherController.cpp
+++ b/server/TetherController.cpp
@@ -696,7 +696,7 @@ int TetherController::setForwardRules(bool add, const char *intIface, const char
         "*raw\n"
         "%s %s -i %s -m rpfilter --invert ! -s fe80::/64 -j DROP\n"
         "COMMIT\n", op, LOCAL_RAW_PREROUTING, intIface);
-    if (iptablesRestoreFunction(V6, rpfilterCmd, nullptr) == -1 && add) {
+    if (iptablesRestoreFunction(V6, rpfilterCmd, nullptr) == -1 && add && false) {
         return -EREMOTEIO;
     }
 
